### ISSUE TITLE
Client crashes when building together

### ISSUE THREAD

**Issue by cablegunmaster** [OpenRCT2/OpenRCT2#8122]
**OS:** Ubuntu 18.04
**Version: 0.2.1
**Commit/Build:** : 38cc469

- [ no] Reproducible in RCT2 (vanilla)?
- [ x ] Specific to multiplayer?

**Steps to reproduce:**
1. Start server
2. get 2 persons to build , he was using mini railway and I was building at the same time a chairlift.
3.  Game crashed.

**Dump file**
[OpenRCT2.zip](https://github.com/OpenRCT2/OpenRCT2/files/2496520/OpenRCT2.zip)

**Screenshots / Video:**

**Save game:**
inside zip file.

**Comment by Broxzier** [comment]
> Assumption: At the exact same time building causes the crash?

I don't think this is the case. When holding a ghost piece, it gets placed and removed every tick on the client, and gets a new free ride ID every time. This avoids two rides using the same ID. When the actual place command gets send to the server, the server assigns the ID, and this matches the clients (who perform the same commands in the same order).

Error: Unhandled exception thrown: read access violation.
gNetwork.player_list._Mypair._Myval2._Myfirst was 0x1110301AD64F7BA.
Callstack:

```
>	openrct2.dll!game_log_multiplayer_command(int command, const int * eax, const int * ebx, const int * ecx, int * edx, int * edi, int * ebp) Line 544	C++
 	openrct2.dll!game_do_command_p(unsigned int command, int * eax, int * ebx, int * ecx, int * edx, int * esi, int * edi, int * ebp) Line 422	C++
 	[Inline Frame] openrct2.dll!game_do_command(int eax, int ebx, int ecx, int edx, int esi, int edi, int ebp) Line 359	C++
 	openrct2.dll!Network::ProcessGameCommandQueue() Line 1615	C++
 	[Inline Frame] openrct2.dll!network_process_game_commands() Line 2785	C++
 	openrct2.dll!OpenRCT2::GameState::UpdateLogic() Line 295	C++
 	openrct2.dll!OpenRCT2::GameState::Update() Line 141	C++
 	openrct2.dll!OpenRCT2::Context::Update() Line 940	C++
 	[Inline Frame] openrct2.dll!OpenRCT2::Context::RunFixedFrame() Line 857	C++
 	[Inline Frame] openrct2.dll!OpenRCT2::Context::RunFrame() Line 830	C++
 	openrct2.dll!OpenRCT2::Context::RunGameLoop() Line 801	C++
 	openrct2.dll!OpenRCT2::Context::RunOpenRCT2(int argc, const char * * argv) Line 216	C++
 	openrct2.dll!NormalisedMain(int argc, const char * * argv) Line 60	C++
 	openrct2.dll!LaunchOpenRCT2(int argc, wchar_t * * argvW) Line 46	C++
```

`gNetwork.player_list` seems to hold 4 objects, but in `game_log_multiplayer_command` somehow a fifth index (4) is used.

**Comment by IntelOrca** [comment]
Did anyone else join the game at the point it crashed?

**Comment by cablegunmaster** [comment]
No ... nobody joined the game at the point it crashed. (as far as I know).

**Comment by ZehMatt** [comment]
This might be related on how players are managed https://github.com/OpenRCT2/OpenRCT2/issues/5684

**Comment by Broxzier** [comment]
Were you playing on the server, or did you connect to it as a client? And which one crashed exactly?

If you have action logging enabled (which looks like you have from the core dump), posting the log of the crash could be helpful. They're located in OpenRCT2/serverlogs/ServerName.

**Comment by cablegunmaster** [comment]
I was connected as an client.  The client crashed / disconnected?, the server was running without issues. (both run same development builds). I can't guarantee the same builds from the other clients though.

[20181019-130848.txt](https://github.com/OpenRCT2/OpenRCT2/files/2498453/20181019-130848.txt)

**Comment by cablegunmaster** [comment]
(https://github.com/OpenRCT2/OpenRCT2/files/2498550/d08ab085-6c2d-4f8b-924f-c0c2a35a7d41.zip)
Crashed happened after placing a "Snow themed ice sign" might not be relevant but happened at around 22:58 . RCT2 crashed Client side only.
